src/vopr_hub/main.go

```diff
@@ -0,0 +1,853 @@
+/*
+Note: The VOPR Hub should not be run from within the tigerbeetle directory
+However, to parse the output correctly the VOPR must run from within the tigerbeetle directory
+To run the VOPR Hub Zig must be installed and five environmental variables are required:
+1. TIGERBEETLE_DIRECTORY where TigerBeetle is stored.
+2. ISSUE_DIRECTORY where issues are stored on disk
+3. DEVELOPER_TOKEN for access to GitHub
+4. VOPR_HUB_ADDRESS for the IP address to listen on for incoming messages
+5. REPOSITORY_URL to post the GitHub issue
+*/
+
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"math"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const MAX_CONCURRENT_CONNECTIONS = 4
+const MAX_QUEUING_MESSAGES = 100
+const LENGTH_OF_VOPR_MESSAGE = 45
+
+// GitHub's hard character limit for issues is 65536
+const MAX_GITHUB_ISSUE_SIZE = 60000
+
+var debug_mode bool
+var max_length_of_vopr_output = 8 * math.Pow(2, 25)
+var tigerbeetle_directory string
+var issue_directory string
+var developer_token string
+var vopr_hub_address string
+var repository_url string
+
+// An alias for the vopr_message's byte array
+type vopr_message_byte_array [LENGTH_OF_VOPR_MESSAGE]byte
+
+// Struct for decoded VOPR message
+// bug type 1 - correctness
+// bug type 2 - liveness
+// bug type 3 - crash
+type vopr_message struct {
+	bug    uint8
+	seed   uint64
+	commit [20]byte
+	hash   [16]byte
+}
+
+// The VOPR's output is stored in a vopr_output struct where certain elements can be extracted and
+// processed.
+// parameters includes information about the conditions under which TigerBeetle is run.
+type vopr_output struct {
+	logs             []byte
+	stack_trace      []byte
+	stack_trace_hash string
+	parameters       string
+	seed_passed      bool
+}
+
+// Checks if the simulator passed or the bug was confirmed.
+func (output *vopr_output) test_passed() bool {
+	passed_regexpr := regexp.MustCompile(`PASSED`)
```

Comment:
@sentientwaffle I'm using strings.Contains here as suggested. Did you want me to move the other regex variable declarations outside of the functions as well to avoid recompiling them? 

So that includes the ones in the extract_parameters and parse_stack_trace functions?

Reply:
No, it is ok as-is.